samples/nrf9160/mqtt_simple/src/main.c

```diff
@@ -509,6 +510,28 @@ static void button_handler(uint32_t button_states, uint32_t has_changed)
 }
 #endif
 
+#if defined(CONFIG_SHELL)
```

Comment:
It's a sample, if it enables shell explicitly, I don't think there's a need to compile this code conditionally.